### PR TITLE
Structured plan renderer: Implement the main functionality for the renderer

### PR DESCRIPTION
**Pull request by liamcervante**
This PR fills in the Render functions on the main Renderer interface of the new renderer.

It also copies the existing diff_test.go file from the old renderer into the new one and verifies that all the tests work.

[To see what it took to make the tests run in the new renderer with this diff.](https://github.com/hashicorp/terraform/commit/a694b64aa3697d4ac208057b5ffd735319f61eb5#diff-8b1fb71dcd519e3c69ce0a6232c7c827d23a79cfa717ff831f5957bdcea8646e)
[To see the changes made to the tests where making the renderers identical wasn't possible see this diff.](https://github.com/hashicorp/terraform/commit/06a3bcd6153766775d846f79fd524ff1208b3507#diff-8b1fb71dcd519e3c69ce0a6232c7c827d23a79cfa717ff831f5957bdcea8646e)

in general, I've tried to make things exactly the same. But some of randomness in the old renderer was impossible to reimplement and the changes to the tests reflect this.

Summaries of the changes I made:

- In the old renderer, JSON objects were rendered as mix between objects and maps, in my renderer they just renderer in the same way as other objects
- the order of additions and deletions in sets in the old renderer was random, the new renderer always shows deletions first and then additions
- the old renderer sometimes did and sometimes didn't show the `-> null` suffix on deleted sensitive values, mine always does. Note, the old one copied whatever the type of value was (so wouldn't for maps, but would for primitives etc). This is very hard for my new renderer to do.
- the old renderer wouldn't treat sensitive values in collections the same as normal values in collections, so a list would show a value that hadn't changed as being created just because it's sensitive status has changed (potentially leaking the sensitive value), and sets would treat sensitive values as being updated when modified instead of the old value deleted and the new value added like normal values. my renderer is much more consistent about this.

This PR also includes mostly minor fixes to indentation, and two big changes around how sensitive blocks and replace paths in blocks work. We were missing functionality for when entire block groups are marked as sensitive or forcing replacements. This PR adds that functionality in by adding additional metadata to the blocks struct in the renderer and allowing the renderer to short circuit this. I do have an idea for how this can be tidied up long term, but I think this should be acceptable short term.

### CODE REVIEW

File: internal/command/jsonformat/computed/renderers/map.go [hunk 1]
```diff
@@ -14,31 +14,44 @@ import (
...
+
 	if len(renderer.elements) == 0 {
-		return fmt.Sprintf("{}%s%s", nullSuffix(opts.OverrideNullSuffix, diff.Action), forcesReplacement(diff.Replace, opts.OverrideForcesReplacement))
+		return fmt.Sprintf("{}%s%s", nullSuffix(opts.OverrideNullSuffix, diff.Action), forcesReplacement(forcesReplacementSelf, opts.OverrideForcesReplacement))
 	}
 
+	maximumKeyLen := 0
+	for key := range renderer.elements {
+		if maximumKeyLen < len(key) {
+			maximumKeyLen = len(key)
```
Comment (alisdair) [inline internal/command/jsonformat/computed/renderers/map.go:50]: Sorry, my brain is fogging up after all this reviewing, but shouldn't this be using `ensureValidAttributeName` like we do with the object renderer?

File: internal/command/jsonformat/computed/renderers/sensitive_block.go [hunk 2]
```diff
@@ -37,15 +37,9 @@ func (renderer sensitiveBlockRenderer) WarningsHuman(diff computed.Diff, indent
 		return []string{}
 	}
 
-	var warning string
 	if renderer.beforeSensitive {
-		warning = fmt.Sprintf("  # [yellow]Warning[reset]: this block will no longer be marked as sensitive\n%s  # after applying this change.", formatIndent(indent))
+		return []string{fmt.Sprintf("  # [yellow]Warning[reset]: this block will no longer be marked as sensitive\n%s  # after applying this change.", formatIndent(indent))}
```
Comment (alisdair) [inline internal/command/jsonformat/computed/renderers/sensitive_block.go:41]: Noted elsewhere in an invisible comment, but for clarity again: we need to colorize these format strings before we interpolate them, not later.

File: internal/command/jsonplan/plan.go [hunk 3]
```diff
@@ -437,17 +458,23 @@ func (p *plan) marshalResourceChanges(resources []*plans.ResourceInstanceChangeS
 	return ret, nil
 }
 
-func (p *plan) marshalOutputChanges(changes *plans.Changes) error {
+// MarshalOutputChanges converts the provided internal representation of
+// Changes objects into the .
```
Comment (alisdair) [inline internal/command/jsonplan/plan.go:462]: This comment trails off.

File: internal/command/jsonformat/computed/renderers/sensitive_block.go (hunk 2 above)
Comment (liamcervante) [reply internal/command/jsonformat/computed/renderers/sensitive_block.go:41]: I think this should be fixed in this PR: https://github.com/hashicorp/terraform/pull/32502

File: internal/command/jsonformat/computed/renderers/map.go (hunk 1 above)
Comment (liamcervante) [reply internal/command/jsonformat/computed/renderers/map.go]: We always escape map keys unlike object keys, so I was handling that part elsewhere. But I think it makes sense to do all the escaping in the same place. So I introduced a new function that does the escaping, `ensureValueAttributeName` calls it and the map renderer calls it directly for every key.

File: internal/command/jsonplan/plan.go (hunk 3 above)
Comment (liamcervante) [reply internal/command/jsonplan/plan.go]: Fixed

**Comment by liamcervante** [comment]
Urgh, the tests are failing in this one now as I preemptively stopped mass calling colorize but haven't merged https://github.com/hashicorp/terraform/pull/32502 yet. The tests for this should start passing again once the other PR is merged.

**Review by alisdair — APPROVED** [review]

**Pull request merged**